pkg/languagedetection/internal/detectors/tracer.go

```diff
@@ -0,0 +1,50 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+package detectors
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/discovery/tracermetadata"
+	model "github.com/DataDog/datadog-agent/pkg/languagedetection/languagemodels"
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+)
+
+// NewTracerDetector is a language Detector that uses
+// data written by the tracer to detect a language
+// for a process.
+func NewTracerDetector() model.Detector {
+	return tracerDetector{
+		hostProc: kernel.ProcFSRoot(),
+	}
+}
+
+type tracerDetector struct {
+	hostProc string
+}
+
+func (i tracerDetector) DetectLanguage(proc model.Process) (model.Language, error) {
+	trMeta, err := tracermetadata.GetTracerMetadata(int(proc.GetPid()), i.hostProc)
+	if err != nil {
+		return model.Language{}, fmt.Errorf("error parsing tracer metadata: %s", err)
```

Comment:
Can we avoid formatting this error and just return `err` since it's going to happen on pretty much every call to the language detector?